### PR TITLE
MGMT-3295: On installation delete previews cluster logs

### PR DESCRIPTION
**Pull request by RazRegev**
We saw in some triaging tickets that when we retrieve failed cluster
logs we can get hosts that were part of the previews installation.
In addition, if a host was re-registered with the same dns, we will
get duplicated log files after the second installation.
To avoid this we can simply delete previews installation log files
just before starting the new installation.

### CODE REVIEW

**Review by filanov — APPROVED** [review]

**Review by filanov — CHANGES_REQUESTED** [review]
what about tests?

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -1160,6 +1160,11 @@ func (b *bareMetalInventory) InstallCluster(ctx context.Context, params installe
 			fmt.Sprintf("Updated status of cluster %s to installing", cluster.Name), time.Now())
 	}()
 
+	// Delete previews installation log files from object storage, if exist.
+	if err := b.clusterApi.DeleteClusterLogs(ctx, &cluster, b.objectHandler); err != nil {
```
Comment (tsorya) [inline internal/bminventory/inventory.go:1164]: Can u please add delete logs_collected_at flag too? Lets clearer all the logs relevant stuff before starting installation

**Comment by avishayt** [comment]
s/previews/previous/

File: internal/cluster/cluster_test.go [hunk 2]
```diff
@@ -2070,3 +2070,75 @@ var _ = Describe("Permanently delete clusters", func() {
...
+			},
+		}
+		Expect(db.Create(&c).Error).NotTo(HaveOccurred())
+		createHostsWithLogs(3, clusterId)
+		mockEvents.EXPECT().AddEvent(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).AnyTimes()
+		mockS3Client.EXPECT().DeleteObject(gomock.Any(), gomock.Any()).Return(false, nil).AnyTimes()
+		mockS3Client.EXPECT().ListObjectsByPrefix(gomock.Any(), gomock.Any()).Return([]string{}, nil).AnyTimes()
+	})
+
+	It("logs flags are cleared test", func() {
```
Comment (filanov) [inline internal/cluster/cluster_test.go:2127]: no bad flows?

File: internal/cluster/cluster.go [hunk 3]
```diff
@@ -674,7 +675,23 @@ func (m *Manager) SetConnectivityMajorityGroupsForCluster(clusterID strfmt.UUID,
 	return nil
 }
 
-func (m *Manager) DeleteClusterLogs(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) error {
+func (m *Manager) ClearClusterLogs(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API, db *gorm.DB) error {
+	if err := m.DeleteClusterLogsFromS3(ctx, c, objectHandler); err != nil {
```
Comment (tsorya) [inline internal/cluster/cluster.go:679]: lets change the order, first delete flags and only then files

File: internal/cluster/cluster.go [hunk 4]
```diff
@@ -674,7 +675,28 @@ func (m *Manager) SetConnectivityMajorityGroupsForCluster(clusterID strfmt.UUID,
 	return nil
 }
 
-func (m *Manager) DeleteClusterLogs(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) error {
+func (m *Manager) ClearClusterLogs(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API, db *gorm.DB) error {
+	resetVal := strfmt.DateTime(time.Time{})
+	for _, h := range c.Hosts {
+		if h.LogsCollectedAt == resetVal {
```
Comment (tsorya) [inline internal/cluster/cluster.go:681]: no need too

File: internal/cluster/cluster.go [hunk 5]
```diff
@@ -674,7 +675,28 @@ func (m *Manager) SetConnectivityMajorityGroupsForCluster(clusterID strfmt.UUID,
...
+			continue
+		}
+		if err := db.Model(h).Update("logs_collected_at", resetVal).Error; err != nil {
+			return err
+		}
+	}
+	if c.ControllerLogsCollectedAt != resetVal {
```
Comment (tsorya) [inline internal/cluster/cluster.go:688]: same

File: internal/cluster/cluster.go [hunk 6]
```diff
@@ -674,7 +675,28 @@ func (m *Manager) SetConnectivityMajorityGroupsForCluster(clusterID strfmt.UUID,
...
+			return err
+		}
+	}
+	if err := m.DeleteClusterLogsFromS3(ctx, c, objectHandler); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (m *Manager) DeleteClusterLogsFromS3(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) error {
```
Comment (filanov) [inline internal/cluster/cluster.go:699]:
```suggestion
func (m *Manager) DeleteClusterLogs(ctx context.Context, c *common.Cluster, objectHandler s3wrapper.API) error {
```

**Comment by RazRegev** [comment]
opened https://github.com/openshift/assisted-service/pull/821

**Pull request closed without merging**